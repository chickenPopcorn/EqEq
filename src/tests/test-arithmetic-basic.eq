BasicMaths = {
  alpha = { 3; }
}

BasicMaths:find alpha {
  print("alpha(%0.f) squared is %0.f\n", alpha, alpha ^ 2);
  print("alpha minus %.0f is %0.f\n", 2, alpha - 2);
  print("alpha plus alpha is %0.f\n", alpha + alpha);
  print("alpha divided by %0.f is %0.2f\n", 2, alpha / 2);
  print("alpha modulo %0.f is %0.f\n", 2, alpha % 2);
  print("negative alpha is %0.f\n", -alpha);

  print("1 is true, 0 is false\n");
  print("alpha negated is %0.f\n", !alpha);
  print("alpha double negated is %0.f\n", !!alpha);

  print("alpha greater than %0.f is %0.f\n", 2, alpha > 2);
  print(
    "alpha greater than or equal to %0.f is %0.f\n",
    2, alpha >= 2);

  print("alpha less than %0.f is %0.f\n", 2, alpha < 2);
  print(
    "alpha less than or equal to %0.f is %0.f\n",
    2, alpha <= 2);

  print(
    "alpha times %0.f plus %0.f is %0.f\n",
    2, 1, alpha * 2 + 1);
}
