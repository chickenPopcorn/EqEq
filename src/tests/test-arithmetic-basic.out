alpha(3) squared is 9
alpha minus 2 is 1
alpha plus alpha is 6
alpha divided by 2 is 1
alpha modulo 2 is 1
negative alpha is -3
alpha negated is 1
alpha double negated is 0
alpha greater than 2 is 1 (1:tr, 0:fls)
alpha greater than or equal to 2 is 1 (1:tr, 0:fls)
alpha less than 2 is 0 (1:tr, 0:fls)
alpha less than or equal to 2 is 0 (1:tr, 0:fls)
alpha times 2 plus 1 is 7
